### PR TITLE
Fix test HTTP proxy server returning early

### PR DESCRIPTION
**Pull request by atburke**
This change fixes a bug in the local HTTP proxy server we use in a few tests. The server's handler would only wait for one side of the proxied connection to finish before returning, which occasionally caused test flake.

Related: #27651

### CODE REVIEW

File: api/testhelpers/proxy.go [hunk 1]
```diff
@@ -86,9 +86,17 @@ func (p *ProxyHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
...
+		select {
+		case <-time.After(10 * time.Second):
+			return
+		case <-errc:
+			numFinished++
+			if numFinished == 2 {
+				return
+			}
+		}
 	}
```
Comment (nklaassen) [inline api/testhelpers/proxy.go:100]: hmm I kinda worry that one of the replicate goroutines can just can stuck in a `io.Copy` call now, but maybe that's not the end of the world, there is a timeout.

But I do wonder, what is the scenario where one side of the connection has closed but we shouldn't close the other side? If that needs to be avoided, I think that would imply there's an issue with lib/utils.ProxyConn as well. Have you been able to reproduce the flake and have you seen improvement with this?

I am more suspicious of the second return from `hj.Hijack()` which we're currently ignoring, it's documented to be a bufio.Reader that may contain unprocessed buffered data from the client.

Comment (atburke) [reply api/testhelpers/proxy.go]: Well, I thought I had reproduced it yesterday by delaying the responding handler, but I tried it again today and it's working fine, so 🤷

I also tried mocking a scenario where the `hj.Hijack()` buffer isn't empty, and that does cause an error. I have no idea if that's what's causing the flakiness (although if it is, there are several more places in Teleport where we're ignoring the hijack buffer that we should fix).

**Review by EdwardDowling — APPROVED** [review]

**Comment by zmb3** [comment]
@atburke what's the status on this one?

**Comment by zmb3** [comment]
@atburke friendly ping - any updates here?

`TestResolveDefaultAddrHTTPProxy` has been flaking for weeks and if we're close to a fix we should wrap it up.

**Comment by atburke** [comment]
@zmb3 Sorry for the delay, I was out of office. I think I've lost confidence that this PR actually addresses the flakiness and may close it. I haven't been able to reproduce the flake locally.

**Comment by zmb3** [comment]
Should we close this one out?

**Pull request closed without merging**